docs/proposals/koordlet/20230728-support-cold-memory-compute.md

```diff
@@ -0,0 +1,217 @@
+
+```go
+// GetMemTotalKB returns the node's memory total quantity (kB)
+func GetMemTotalKB() (int64, error) {
+	return usage, nil
+}
+
+// GetMemFreeKB returns the node's memory free quantity (kB)
+func GetMemFreeKB() (int64, error) {
+	return usage, nil
+}
+```
+
+Define a function named CollectNodeMemoryWithHotPageUsage().  Compute node memory usage with hot page and generate metric. collectNodeResUsed() function will call that and insert memoryWithHotPage metric.
+
+```go
+func (n *nodeResourceCollector) CollectNodeMemoryWithHotPageUsage(collectTime time.Time) (metriccache.MetricSample, error) {
+	return nodeMemorWithHotPageyUsageMetric, nil
+}
+```
+
+The same process is executed in podresource collector to collect idle page. Do not repeat.
+
+#### Report memory usage including hot page
+
+![image](../../images/support-cold-memory-2.svg)
+
+collectNodeMetric() is used to query metirc and return CPU And MemUsed in pkg/koordlet/statesinformer/impl/states_nodemetirc.go file.
+
+We can report memory usage including hot page in collectNodeMetric().
+
+
+```go
+func (r *nodeMetricInformer) collectNodeMetric(queryparam metriccache.QueryParam) (corev1.ResourceList, time.Duration, error) {
+	rl[corev1.ResourceCPU] = *resource.NewMilliQuantity(int64(cpuUsed*1000), resource.DecimalSI)
+	rl[corev1.ResourceMemory] = *resource.NewQuantity(int64(memUsed), resource.BinarySI)
+	return rl, cpuAggregateResult.TimeRangeDuration(), nil
+}
+```
+
+The same process is executed pod informer to report memory usage. Do not repeat.
+
+#### Define cold memory API
+
+provide memory collect policy access
+
+Add field named MemoryCollectPolicy to represent which method is used to collect memory usage. Such as usageWithHotPageCache, usageWithoutPageCache,  usageWithPageCache.
+
+You can create a crd nodemeric resource and specify value of spec.CollectPolicy.MemoryCollectPolicy to start collect cold memory compute. 
+
+```go
+// NodeMetricCollectPolicy defines the Metric collection policy
+type NodeMetricCollectPolicy struct {
+	// AggregateDurationSeconds represents the aggregation period in seconds
+	AggregateDurationSeconds *int64 `json:"aggregateDurationSeconds,omitempty"`
+	// ReportIntervalSeconds represents the report period in seconds
+	ReportIntervalSeconds *int64 `json:"reportIntervalSeconds,omitempty"`
+	// NodeAggregatePolicy represents the target grain of node aggregated usage
+	NodeAggregatePolicy *AggregatePolicy `json:"nodeAggregatePolicy,omitempty"`
+	//MemoryWithHotPageCollectPolicy represents whether collet memory usage with hot page instead of memory usage without page cache
+	MemoryCollectPolicy string `json:"memoryCollectPolicy,omitempty"`
+}
+```
+## Alternatives
+
+**kidled**: an open source cold memory collection solution.
+
+https://github.com/alibaba/cloud-kernel/blob/linux-next/Documentation/vm/kidled.rst
+
+**kstaled**:proactively reclaiming idle memory
```

Comment:
Need a brief introduction about the other cold page techniques. Otherwise, it is hard to know their differences.